querybook/webapp/components/QueryComposer/RunQuery.tsx

```diff
@@ -46,13 +43,9 @@ export async function transformQuery(
         sampleRate
     );
 
-    const limitedQuery = await transformLimitedQuery(
-        sampledQuery,
-        rowLimit,
-        engine
-    );
+    await checkUnlimitedQuery(sampledQuery, rowLimit, engine);
```

Comment:
what about still keeping the old logic here, but only replace the `transformLimitedQuery` with the backend api, like sampledQuery does? you probably thought about this, what is the concern?

btw, I think we should keep the logic consistent for both the sampling and limited query

Reply:
I had two thoughts:

1. There would be three API calls required to prep a query before sending it to be executed, seemed like too much.  I was thinking about creating a single unified endpoint that combined templating/sampling/limiting, but that was even more changes
2. Scheduled DataDocs don't go through that flow, so it doesn't make it simpler; it just adds a new endpoint purely for adhoc executions

That said, I'm fine with refactoring to use a new endpoint for this.  If it returned a flag for unlimited queries, then we can remove even more frontend code.

Reply:
you raised a good point. I was also thinking of merging all the transforms into one single API endpoint. maybe we could try that route?

Reply:
I combined sampling/limiting into `POST /query/transform/`, but left the individual endpoints&mdash;sampling preview uses `/query/transform/sampling/`, and there may be some use for a standalone limited transform as well.